presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -38,4 +61,680 @@ public static JsonGenerator createJsonGenerator(JsonFactory factory, SliceOutput
+                throws IOException;
+
+        public static JsonGeneratorWriter createJsonGeneratorWriter(Type type)
+        {
+            String baseType = type.getTypeSignature().getBase();
+            if (type instanceof UnknownType) {
+                return new JsonGeneratorUnknownWriter();
+            }
+            else if (baseType.equals(StandardTypes.BOOLEAN)) {
+                return new JsonGeneratorBooleanWriter(type);
+            }
+            else if (baseType.equals(StandardTypes.TINYINT) ||
+                    baseType.equals(StandardTypes.SMALLINT) ||
+                    baseType.equals(StandardTypes.INTEGER) ||
+                    baseType.equals(StandardTypes.BIGINT)) {
+                return new JsonGeneratorLongWriter(type);
+            }
+            else if (baseType.equals(StandardTypes.REAL)) {
+                return new JsonGeneratorRealWriter(type);
+            }
+            else if (baseType.equals(StandardTypes.DOUBLE)) {
+                return new JsonGeneratorDoubleWriter(type);
+            }
+            else if (baseType.equals(StandardTypes.DECIMAL)) {
+                if (isShortDecimal(type)) {
+                    return new JsonGeneratorShortDecimalWriter((DecimalType) type);
+                }
+                else {
+                    return new JsonGeneratorLongDeicmalWriter((DecimalType) type);
+                }
+            }
+            else if (baseType.equals(StandardTypes.VARCHAR)) {
+                return new JsonGeneratorVarcharWriter(type);
+            }
+            else if (baseType.equals(StandardTypes.JSON)) {
+                return new JsonGeneratorJsonWriter(type);
+            }
+
+            if (type instanceof ArrayType) {
+                return new JsonGeneratorArrayWriter((ArrayType) type);
+            }
+            if (type instanceof MapType) {
+                return new JsonGeneratorMapWriter((MapType) type);
+            }
+            if (type instanceof RowType) {
+                return new JsonGeneratorRowWriter((RowType) type);
+            }
+
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Unsupported type: %s", type));
+        }
+
+        private static class JsonGeneratorUnknownWriter
+                extends JsonGeneratorWriter
+        {
+            @Override
+            public void writeJsonValue(JsonGenerator jsonGenerator, Block block, int position, ConnectorSession session)
+                    throws IOException
+            {
+                jsonGenerator.writeNull();
+            }
+
+            @Override
+            public void writeObjectKeyValuePair(JsonGenerator jsonGenerator, String objectKey, Block block, int position, ConnectorSession session)
+                    throws IOException
+            {
+                jsonGenerator.writeNullField(objectKey);
```

Comment:
Would it be much easier if you do `writeFieldName` and `writeNull/writeBoolean/writeNumber/...`? This way, you wouldn't need two methods. Am I understanding this correctly?